Jenkinsfile

```diff
@@ -13,10 +13,54 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+<<<<<<< HEAD
+podTemplate(yaml: '''
+apiVersion: v1
+kind: Pod
+metadata: 
+    name: dind 
+spec:
+  affinity:
+    nodeAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 1
+        preference:
+          matchExpressions:
+          - key: palisade-node-name
+            operator: In
+            values: 
+            - node1
+            - node2
+            - node3
+  containers:
+  - name: jnlp
+    image: jenkins/jnlp-slave
+    imagePullPolicy: Always
+    args: 
+    - $(JENKINS_SECRET)
+    - $(JENKINS_NAME)
+    resources:
+      requests:
+        ephemeral-storage: "4Gi"
+      limits:
+        ephemeral-storage: "8Gi"
 
-podTemplate(containers: [
-        containerTemplate(name: 'maven', image: 'maven:3.6.1-jdk-11', ttyEnabled: true, command: 'cat')
-]) {
+  - name: docker-cmds
+    image: 779921734503.dkr.ecr.eu-west-1.amazonaws.com/jnlp-did:INFRA
```

Comment:
Does this need to specify the ECR? If so can we parameterise this?

Reply:
Since we are using a real Kubernetes installation rather than microk8s, it is necessary to specify the full tag for the ECR.
I have tried parameterising the variable using ${}, ${env.}, $(), $(env.) - none of which result in the substitution working at all.
The intention, is to incorporate the pod template definition into the Jenkins template longer term so that the Jenkinsfiles are sparse, so the visibility of the ECR registry should not remain an issue.
I can spend more time looking at this but it is not as obvious as it would first appear.